db/sqlglue.c

```diff
@@ -4993,15 +4993,6 @@ int sqlite3BtreeBeginTrans(Vdbe *vdbe, Btree *pBt, int wrflag, int *pSchemaVersi
     }
 #endif
 
-    struct dbtable *db =
-        &thedb->static_table; 
-    /* Latch last commit LSN */
-    if (clnt->dbtran.mode == TRANLEVEL_MODSNAP && !clnt->modsnap_in_progress && (db->handle != NULL) &&
-        (populate_modsnap_state(clnt) != 0)) {
-        rc = SQLITE_INTERNAL;
-        goto done;
-    }
```

Comment:
Issue was that we already hold table locks at this point, and `populate_modsnap_state` then tried to grab the schema lock